src/Models/IoTContainerizedProject.ts

```diff
@@ -167,13 +130,14 @@ export class IoTContainerizedProject extends IoTWorkbenchProjectBase {
    * environment.
    */
   async openProject(
-      projectPath: string, openInNewWindow: boolean,
-      openScenario: OpenScenario) {
-    if (!FileUtility.directoryExists(ScaffoldType.Local, projectPath)) {
-      channelShowAndAppendLine(
-          this.channel, `Can not find project path ${projectPath}.`);
-      return;
+      scaffoldType: ScaffoldType, openInNewWindow: boolean,
+      openScenario: OpenScenario): Promise<void> {
+    if (!await FileUtility.directoryExists(

```

Comment:
The root path validation logic is spread in many function, can we abstract it as a common function to validateRootPath?

Reply:
Fixed.